run/awake_start

```diff
@@ -1,21 +1,40 @@
 #!/bin/bash -eu
 
-if [ ! -x /root/bin/tesla_api.py ] || [ ! -s /mutable/cache.json ]
+if ( [ ! -x /root/bin/tesla_api.py ] || [ ! -s /mutable/cache.json ] ) && [[ ( ! -n "${TESLAFI_API_TOKEN:+x}" ) ]]
 then
-  log "not keeping car awake."
+  log "Not keeping car awake."
   exit
 fi
 
 function ping {
   while true
   do
-    if /root/bin/tesla_api.py streaming_ping
+    if [[ ( -n "${TESLA_REFRESH_TOKEN:+x}" ) ]]
+    # Use Tesla API to keep car awake
     then
-      sleep 90
-    else
-      log "failed to contact car, retrying"
-      sleep 5
+      if /root/bin/tesla_api.py streaming_ping
+      then
+        log "Contact car every 90s using Tesla API to keep it awake."
+        sleep 90
+      else
+        log "Failed to contact car using Tesla API, retrying in 5s..."
+        sleep 5
+      fi
     fi
+
+    if [[ ( -n "${TESLAFI_API_TOKEN:+x}" ) ]]
+    # Use TeslaFi API to keep car awake
+    then
+      if [[ $(curl -s -H "Authorization: Bearer $TESLAFI_API_TOKEN" "https://www.teslafi.com/feed.php?command=wake") = "Wake command received" ]]
+      then
+        # TeslaFi's default sleep timer is 15min (900s). No need to contact too often to prevent API lockout (max 3 calls/min).
```

Comment:
does this also mean that the car will be kept awake a minimum of 15 minutes when using TeslaFi?

Reply:
No, it's actually 30 min in practice. 15 min. is how long TeslaFi will stop polling the car. This is the general behavior and limitation for any 3rd-party service provider like TeslaFi and others. This is a good read on how that works, if you'r curious:
https://support.teslafi.com/knowledge-bases/2/articles/640-enabling-sleep-settings-to-limit-vampire-loss